app/services/support_interface/application_monitor.rb

```diff
@@ -0,0 +1,27 @@
+module SupportInterface
+  class ApplicationMonitor
+    def applications_to_disabled_courses
+      active_applications.where(
+        course_option: CourseOption.joins(:course).where('courses.open_on_apply' => false),
+      ).map(&:application_form)
+    end
+
+    def applications_to_hidden_courses
+      active_applications.where(
+        course_option: CourseOption.joins(:course).where('courses.open_on_apply' => true, 'courses.exposed_in_find' => false),
+      ).map(&:application_form)
+    end
+
+    def applications_to_full_courses
```

Comment:
I think we have a couple of conflicting concepts here. In hindsight I should've given `invalidated_by_find` a more specific name, but what that field indicates currently is whether a site is still a valid selection for the chosen course, or if it's been removed. That's distinct from whether or not the site is full, which is indicated by `vacancy_status: :no_vacancies`.

"full_courses" is probably best described as any course where all course options are `vacancy_status: :no_vacancies`. At a simpler level, maybe we could make this `applications_to_full_sites` which would query:

```
      active_applications.where(
        course_option: CourseOption.where(vacancy_status: 'no_vacancies'),
      ).map(&:application_form)
```

Reply:
O wow, great catch. I've changed it up here: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1834/commits/4746c8823e2f09ef1a2a338f69a0897400b32648